packages/components/src/Components/FilterHooks/constants.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Badge, Tooltip } from '@patternfly/react-core';
+
+export function mapGroups(currSelection, valuesKey = 'values') {
+    return Object.entries(currSelection || {}).map(([ groupKey, groupValue ]) => {
+        const values = constructValues(groupValue, groupKey);
+        if (values?.length > 0) {
+            return {
+                type: 'tags',
+                key: groupKey,
+                category: values[0]?.group?.value || values[0]?.group?.label,
+                [valuesKey]: values
+            };
+        }
+    }).filter(Boolean);
+}
+
+export function constructValues(groupValue, groupKey) {
+    return Object.entries(groupValue || {}).map(([ key, { isSelected, group, value, item }]) => {
+        if (isSelected) {
+            const { key: tagKey, value: tagValue } = item?.meta?.tag || {
+                key: groupKey,
+                value: value || item?.tagValue
+            };
+            return {
+                key,
+                tagKey,
+                value: tagValue,
+                name: `${tagKey}${tagValue ? `=${tagValue}` : ''}`,
+                group: { value: groupKey, ...group }
+            };
+        }
+    }).filter(Boolean);
```

Comment:
Both map and filter actions could be done in a single `reduce` iteration. It should make it slightly quicker.